core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -1322,6 +1326,30 @@ class KafkaApis(val requestChannel: RequestChannel,
     txnCoordinator.handleInitPid(initPidRequest.transactionalId, sendResponseCallback)
   }
 
+  def handleEndTxnRequest(request: Request): Unit = {
+    throw new UnsupportedOperationException("Not implemented yet.")
+  }
+
+  def handleAbortTxnRequest(request: Request): Unit = {
+    throw new UnsupportedOperationException("Not implemented yet.")
+  }
+
+  def handleAddPartitionToTxnRequest(request: Request): Unit = {
+    throw new UnsupportedOperationException("Not implemented yet.")
+  }
+
+  def handleAddOffsetsToTxnRequest(request: Request): Unit = {
+    throw new UnsupportedOperationException("Not implemented yet.")
+  }
+
+  def handleWriteTxnMarkerRequest(request: Request): Unit = {
+    throw new UnsupportedOperationException("Not implemented yet.")
+  }
+
+  def handleTxnOffsetCommitRequest(request: Request): Unit = {
+    throw new UnsupportedOperationException("Not implemented yet.")
```

Comment:
In KAFKA-4954, each request handler call is changed to the following before a response is sent. It's much easier to add such logic if the request handler sends a response instead of throwing an exception. Otherwise, anyone who removes the UnsupportedOperationException has to remember adding that logic later on.

```
      def createResponse(throttleTimeMs: Int): AbstractResponse = new UpdateMetadataResponse(Errors.CLUSTER_AUTHORIZATION_FAILED)
     sendResponseMaybeThrottle(request, createResponse)
```